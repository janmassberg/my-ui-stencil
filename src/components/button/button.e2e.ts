import { newE2EPage } from "@stencil/core/testing";

describe("my-ui-button", () => {
  it("should have variant classNames and render button with aria-label attribute", async () => {
    const page = await newE2EPage();
    await page.setContent(
      '<my-ui-button label="Label" kind="primary">Content</my-ui-button>',
    );
    const el = await page.find("my-ui-button");
    await page.waitForChanges();
    expect(el).toHaveClass("hydrated");
    expect(el).toMatchSnapshot();
  });
});
