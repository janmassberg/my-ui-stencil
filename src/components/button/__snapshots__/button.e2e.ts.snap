// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my-ui-button should have variant classNames and render button with aria-label attribute 1`] = `
<my-ui-button class="hydrated my-ui-button my-ui-button--primary my-ui-button--size-m" kind="primary" label="Label">
  <!---->
  <button aria-label="Label" class="my-ui-button__button">
    Content
  </button>
</my-ui-button>
`;
